src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,965 @@
+============== file: Google/Example/Library/V1/ResourceNames.cs ==============
+// Copyright 2016, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated code. DO NOT EDIT!
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Example.Library.V1
+{
+    /// <summary>
+    /// Resource name for the 'shelf' resource.
+    /// </summary>
+    public sealed partial class ShelfName : IResourceName, IEquatable<ShelfName>
+    {
+        private static readonly PathTemplate s_template = new PathTemplate("shelves/{shelf_id}");
+
+        /// <summary>
+        /// Parses a shelf resource name in string form into a <see cref="ShelfName"/>.
+        /// </summary>
+        /// <param name="shelfName">The shelf resource name in string form. Must not be <c>null</c>.</param>
+        /// <returns>The parsed <see cref="ShelfName"/> if successful.</returns>
+        public static ShelfName Parse(string shelfName)
+        {
+            GaxPreconditions.CheckNotNull(shelfName, nameof(shelfName));
+            TemplatedResourceName resourceName = s_template.ParseName(shelfName);
+            return new ShelfName(resourceName[0]);
+        }
+
+        /// <summary>
+        /// Tries to parse shelf resource name in string form into a <see cref="ShelfName"/>/
```

Comment:
Would be nicer as "Tries to parse a" but then we need to work out a/an.
Mind you, that exists for the "into a" as well.
Non-blocking, certainly.